src/base/registrations/registrar.ts

```diff
@@ -95,6 +99,9 @@ async function commitAndRegister(name: string, owner: string, config: Config) {
   let salt = ethers.utils.randomBytes(32);
   let minAge = (await registrar(config).minCommitmentAge()).toNumber();
   let fee = await registrationFee(config);
+  // Avoids gas spent by the owner, trying to commit to a name and not having
+  // enough RAD balance
+  assertBalance((await radToken(config).balanceOf(owner)).toNumber(), fee);
```

Comment:
I think this would be better handled by throwing a custom error that we can catch in the view, see line 94 for eg. Perhaps adding a `Failure.InsufficientBalance` variant to the enum is a good idea.

Reply:
I agree, eventually the `commitAndRegister` function could throw a custom `InsufficientBalance` error and the `registerName` function could re throw this custom error with a custom message, something like `Not enough RAD funds` or in its absence throw a generic `TransactionFailed` error.